docs/content/preview/explore/joins/joins-ysql.md

```diff
@@ -0,0 +1,233 @@
+         Sort Method: quicksort  Memory: 27kB
+         ->  Seq Scan on scores (actual time=1001.133..1001.152 rows=47 loops=1)
+               Remote Filter: (score > 70)
+               Storage Table Read Requests: 1
+               Storage Table Read Execution Time: 1001.020 ms
+ Planning Time: 0.203 ms
+ Execution Time: 2002.086 ms
+ Storage Read Requests: 2
+ Storage Read Execution Time: 2001.631 ms
+ Storage Execution Time: 2001.631 ms
+ Peak Memory Usage: 118 kB
+```
+
+## Hash Join
+
+The Hash Join algorithm starts by hashing the join key columns of both input tables. The hash function transforms the join key values into a hash code. Hash Joins are suitable for equi-joins (joins based on equality) and are effective when the join keys are not sorted or indexed.
+
+To fetch all students who have scored more than `70` in any subject and use Hash join, you would execute,
+
+```sql
+explain (analyze, dist, costs off)
+/*+
+    hashjoin(students scores)
+    set(yb_enable_optimizer_statistics on)
+*/
+SELECT name, subject, score FROM students JOIN scores USING(id) WHERE score > 70;
+```
+
+The query plan would be similar to:
+
+```sql
+ Hash Join (actual time=2002.455..2002.516 rows=47 loops=1)
+   Hash Cond: (scores.id = students.id)
+   ->  Seq Scan on scores (actual time=1001.456..1001.484 rows=47 loops=1)
+         Remote Filter: (score > 70)
+         Storage Table Read Requests: 1
+         Storage Table Read Execution Time: 1001.324 ms
+   ->  Hash (actual time=1000.968..1000.969 rows=20 loops=1)
+         Buckets: 1024  Batches: 1  Memory Usage: 9kB
+         ->  Seq Scan on students (actual time=1000.906..1000.919 rows=20 loops=1)
+               Storage Table Read Requests: 1
+               Storage Table Read Execution Time: 1000.814 ms
+ Planning Time: 0.280 ms
+ Execution Time: 2002.591 ms
+ Storage Read Requests: 2
+ Storage Read Execution Time: 2002.138 ms
+ Storage Execution Time: 2002.138 ms
+ Peak Memory Usage: 133 kB
+```
+
+## Batched Nested Loop (BNL)
+
+In the case of the Nested loop joins, the inner table is accessed multiple times one for each outer table row. This would lead to multiple RPC requests across the different nodes in the cluster, making this join strategy very slow as the outer table gets larger.
+
+To reduce the number of requests sent across the nodes during the Nested Loop join, YugabyteDB adds new optimization to batch multiple keys of the outer table into one RPC request. This batch size can be controlled using the GUC variable `yb_bnl_batch_size`, which defaults to `1` (which effectively means that the feature is `OFF`). Suggested value for this variable is `1024`.
+
+The optimizer will automatically adopt the batch size when other join strategies are not fit for the current query.
+
+Now to fetch all students who have scored more than `70` in any subject using Batched nested loop join, you would execute,
+
+```sql
+explain (analyze, dist, costs off)
+/*+
+    ybbatchednl(students scores)
```

Comment:
This hint is not available below 2.21 so would this be confusing to readers who are not on that version?

Reply:
Okay will make a note of it.

Reply:
removed